app/api/v2/handlers/base_api.py

```diff
@@ -0,0 +1,20 @@
+import abc
+import logging
+
+
+class BaseApi(abc.ABC):
+    def __init__(self, logger=None):
+        self._logger = logger or self._create_logger()
+
+    @property
+    def logger(self):
+        return self._logger
+
+    @abc.abstractmethod
+    def add_routes(self, app):
+        pass
+
+    def _create_logger(self):
+        """"""
+        logger_name = f"{self.__module__}.{self.__class__.__name__}"
```

Comment:
I'm used to logger hierarchies reflecting the package structure more (a lot of module-level `getLogger(__name__)` calls). It seems like caldera uses functional areas for logger names, so I might change this so the default logger is `caldera.api.v2` or something. I do wish our loggers started with `caldera.`

Reply:
I ended up just reusing the v1 logger since it's named "rest_api" and this is the same functional area, just a different api version.